ts/nni_manager/training_service/reusable/trialDispatcher.ts

```diff
@@ -279,64 +296,113 @@ class TrialDispatcher implements TrainingService {
                     this.log.info(`TrialDispatcher: GPU scheduler is enabled.`)
                     this.enableGpuScheduler = true;
                 }
-                this.runnerSettings.enableGpuCollector = this.enableGpuScheduler;
 
-                this.runnerSettings.command = this.trialConfig.command;
                 // Validate to make sure codeDir doesn't have too many files
                 await validateCodeDir(this.trialConfig.codeDir);
                 break;
+            case TrialConfigMetadataKey.PLATFORM_LIST: {
+                const platforms: string[] = value.split(",");
+                for(const platform of platforms) {
+                    let environmentService: EnvironmentService;
+                    switch(platform) {
+                        case 'local':
```

Comment:
This logic is not necessary like this, it should be initialized like a factory pattern.

Reply:
updated, add environmentServiceFactory class to genreate environmentService.